XenAdmin/Wizards/NewNetworkWizard_Pages/NetWDetails.cs

```diff
@@ -172,7 +172,11 @@ private List<int> GetVLANList(PIF nic)
 
             foreach (PIF pif in nic.Connection.Cache.PIFs)
             {
-                if (pif.device == nic.device)
+                XenAPI.Network network = pif.Connection.Resolve(pif.network);

```

Comment:
This should go inside the `if (pif.device == nic.device)` block, because there is no point in resolving the network for all the PIFs, we onlt need to do it for the one that satisfy this condition.